docs/src/architecture.md

```diff
@@ -1,51 +1,176 @@
 # License Manager Architecture
-The License Manager is based on a client/backend architecture. The backend consists of a RESTful API built with Python over a
-PostgreSQL database. The client consists of a timed reconcile job that runs on the control node in the cluster and
-a prolog integration to Slurm.
+`License Manager` works in a client/server architecture, where the main components are the `Agent`, the `API` and the `CLI`.
+
+The `License Manager Agent` is the component that syncs the local `Slurm` license counters with the License Server data.
+It's also responsible for making `Booking` requests to the License Manager API when a job is submitted to Slurm.
+
+`The License Manager API` provides a `RESTful API` where license usage and `Bookings` are tracked. The `License Manager Agent` uses this `API`
+to store the license usage information and to process the `Booking` requests.
+
+The `License Manager CLI` is an interface to interact with the `API` to add new license configurations and to check the usage information for each tracked license.
+
 
 ## License Manager Agent
-The `License Manager Agent` is responsible for keeping the local cluster license totals
-in sync with the the 3rd party license server totals. It's also responsible for making booking requests
-to the `License Manager API` when Slurm is configured to use the `PrologSlurmctld` script provided by `License Manager Agent`.
+`License Manager` is used to ensure that when a job is executed, there are enough free licenses for it to succeed. The `License Manager Agent` is the component that makes it possible.
+
+The agent has two main functionalities:
+
+* Sync the local `Slurm` license counters with the usage from the `License Server`
+
+* Manage the license requests from jobs submitted to the cluster using `Bookings`.
```

Comment:
So, despite the sentences are clear enough, I feel they could be improved somehow to be more "dumb friendly" (I'm one regarding the license manager architecture, that's why I say that)